db/migrate/20240108150948_add_sort_index_to_course.rb

```diff
@@ -0,0 +1,13 @@
+class AddSortIndexToCourse < ActiveRecord::Migration[7.0]
+  class Course < ApplicationRecord
+  end
+
+  def change
+    add_column :courses, :sort_index, :integer, default: 0
+
+    Course
+      .order(name: :asc)
+      .find_each
```

Comment:
@harigopal 

`find_each` does not retain the order from `order(name: :asc)` it updates the `sort_index` based on PK `id`.

If we want to add sort_index based on ascending order of name, then:

- Use `each` instead of `find_each` as I think the number of the course records will not be too many to cause issue.
- Write a custom batch loader.

What is your take ?

Reply:
Good catch. I've replaced `find_each.with_index` with a simple `each_with_index`.